src/createServer.js

```diff
@@ -4,0 +1,113 @@
+/* eslint-disable max-len */
+const http = require('http');
+const { convertToCase } = require('./convertToCase');
+
+function validateRequest(originalText, targetCase) {
+  const errors = [];
+
+  if (!originalText) {
+    errors.push({
+      message:
+        'Text to convert is required. Correct request is: "/<TEXT_TO_CONVERT>?toCase=<CASE_NAME>".',
+    });
+  }
+
+  if (!targetCase) {
+    errors.push({
+      message:
+        '"toCase" query param is required. Correct request is: "/<TEXT_TO_CONVERT>?toCase=<CASE_NAME>".',
+    });
+  } else if (
+    !['SNAKE', 'KEBAB', 'CAMEL', 'PASCAL', 'UPPER'].includes(targetCase)
+  ) {
+    errors.push({
+      message:
+        'This case is not supported. Available cases: SNAKE, KEBAB, CAMEL, PASCAL, UPPER.',
+    });
+  }
+
+  return errors;
+}
+
+function parseUrl(req) {
+  const url = new URL(req.url, `http://${req.headers.host}`);
```

Comment:
Make sure to handle the case where `req.url` is not a valid URL. The `URL` constructor will throw a `TypeError` if the input is not a valid URL. You can use a try-catch block to catch this error and return a 400 response.